server/api/messaging.py

```diff
@@ -0,0 +1,207 @@
+        message_info['conversation'] = Conversation.get_conversation(user, receiver)
+    else:
+        message_info['conversation'] = Conversation.create(**{'user_one': user, "user_two": receiver})
+    new_message = Message.create(**message_info)
+    message_info['conversation'].add_to_relationship('messages', new_message)
+    response_dict['status'] = 201
+    response_dict['message'] = "Successfully sent message to %s" % (message_info['receiver'].profile.name)
+    return jsonify(response_dict), 201
+    
+
+@messaging_views.route("/get_received_messages", methods=["GET"])
+@authenticate
+def received_messages(**kwargs):
+    response_dict = {'messages': []}
+    user = kwargs['user']
+    # Try to get all received messages
+    # if no instances are found, catch 
+    # the exception and tell the user 
+    # there are no messages
+    try:
+        messages = Message.get_instance(multiple=True, **{'receiver': user})
+    except NoResultFound:    
+        response_dict['status'] = 204
+        response_dict['message'] = "You haven't received any messages yet"
+        return jsonify(response_dict), 204
+
+    for message in messages:
+        message_dict = message.to_dict(exclude=['receiver', 'sender', 'created_at'])
+        message_dict['sender'] = message.sender.profile.name
+        message_dict['sender_image_url'] = message.sender.profile.profile_image
+        for field, value in message.get_formatted_info.items():
+            message_dict[field] = value
+        response_dict['messages'].append(message_dict)
+
+    return jsonify(response_dict), 200
+
+@messaging_views.route('/get_sent_messages', methods=["GET"])
+@authenticate
+def sent_messages(**kwargs):
+    response_dict = {'messages': []}
+    user = kwargs['user']
+    try:
+        messages = Message.get_instance(multiple=True, **{'sender': user})
+    except NoResultFound:    
+        response_dict['status'] = 204
+        response_dict['message'] = "You haven't sent any messages yet"
+        return jsonify(response_dict), 204
+
+    for message in messages:
+        message_dict = message.to_dict(excludes=['sender', 'receiver', 'created_at'])
+        for field, value in message.get_formatted_info.items():
+            message_dict[field] = value
+        message['receiver'] = message.receiver.profile.name
+        message['receiver_image'] = message.receiver.profile.profile_image
+        response_dict['messages'].append(message_dict)
+    return jsonify(response_dict), 200
+
+
+@messaging_views.route('/get_conversation_messages', methods=["GET"])
+@authenticate
+def conversation_messages(**kwargs):
+    """
+    Get all messages between the logged in user
+    and the other user then order them by the 
+    created_at field to ensure a proper descending
+    order so the newest messages are at the bottom
+    """
+    response_dict = {'messages': []}
+    user = kwargs['user']
+    other_user_id = request.get_json().get('other_user_id')
```

Comment:
This shouldn't be necessary. We can fetch conversations by their id

Reply:
Just so I'm completely clear, you're saying to replace this with something like `conversation_id = request.get_json().get("conversation_id")` then use that in the query instead right?
Then that should make the get_conversation method obsolete, I'll change that